### DISCUSSION TITLE
Main View goes blank when I change the option of a select input type.

### DISCUSSION THREAD

**Discussion by orphikkkk** [livewire/livewire#5732] [1. Help] [answered]
I am pretty new at Livewire. I have a home page where you can choose the branch of a restaurant, time-slot of a reservation, and the main dish. The select input type has different limit value based on the selected option which will be emitted to a count component. 

I have created a Select input type for the branch and have wire:model  and wire:change to link with the component.
```php
<div class="col-6 col-sm-6">
        <div class="mb-4">
        <label class="form-label">Select Time Slot</label>
            <select class="form-select" aria-label="Default select example"
                    name="time_slot" wire:change="selectedTime()" wire:model="selectedTimeSlotId">
                <option selected>Select Time Slot</option>
                @foreach($timeSlots as $slot)
                      <option wire:key="time-{{$slot->id}}" value="{{$slot->id}}">{{$slot->formatted_start_time}} - {{$slot- 
                            >formatted_end_time}}</option>
                 @endforeach
               </select>
           </div>
 </div>
```
The wire:change function should emit the availability value to the count component which will use it as a limit value for its counter.
```php
 public function selectedTime()
    {
        $timeslot = TimeSlot::has('timeSlotDishes')->find($this->selectedTimeSlotId);
        if ($timeslot)
            $this->availability = $timeslot->timeSlotDishes->availability;
        $this->emit('timeSelected', $this->availability);
    }
```
  But whenever the counter components are rendered in the main component the whole view goes blank. Even when I change the value of another select input (branch) which is not related to the child components the view goes blank

![2023-06-08 3 14 45 PM](https://github.com/livewire/livewire/assets/12593145/84c39a2e-7980-4c9d-9c76-c94140feb608)

**Comment by DaviMenezes** [comment]
check if page contain only 1 element root

**Reply by orphikkkk** [reply]
I have confirmed that It is only one root there is no error message on the console either.

**Comment by tomphilpotts** [comment]
What happens in the network tab?

**Comment by tomphilpotts** [comment]
Try this:
```
    public function updatedSelectedTimeSlotId()
    {
         $timeSlot = TimeSlot::with('timeSlotDishes')
        ->find($this->selectedTimeSlotId);
    
    $this->availability = $timeSlot
        ? $timeSlot->timeSlotDishes->availability
        : null;

    $this->emit('timeSelected', $this->availability);
    }
```

And remove the `wire:change="selectedTime()"`,  instead have the hook work in the component class instead.

**Comment by orphikkkk** [comment, marked as answer]
I couldn't figure out why the issue was caused but when I removed a comment 
`<!-- Main -->`
form the main component the issue was solved. Please do tell me if anybody has an explanation for this.
